### DISCUSSION TITLE
how can i reduce or turn off the black background in shadcn Dialog component

### DISCUSSION THREAD

**Discussion by nikhilmohanv** [shadcn-ui/ui#3765] [Q&A]
![image](https://github.com/shadcn-ui/ui/assets/87137042/8c7e6dfb-653f-4a72-9d8a-a7c9945bbb73)
as in the image the dialogue box have a black background. I tried to reduce background color of dialogue component but it is not working

**Comment by mohd-khan09** [comment]
in component change the background opacity , here i an example where you can set it as a prop and from dailog content import * as React from "react";
import * as DialogPrimitive from "@radix-ui/react-dialog";
import { Cross2Icon } from "@radix-ui/react-icons";

import { cn } from "@/lib/utils";

const Dialog = DialogPrimitive.Root;

const DialogTrigger = DialogPrimitive.Trigger;

const DialogPortal = DialogPrimitive.Portal;

const DialogClose = DialogPrimitive.Close;

type ExtendedDialogOverlayProps = React.ComponentPropsWithoutRef<
  typeof DialogPrimitive.Overlay
> & {
  backgroundOpacity?: number;
};
const DialogOverlay = React.forwardRef<
  React.ElementRef<typeof DialogPrimitive.Overlay>,
  ExtendedDialogOverlayProps
>(({ className, backgroundOpacity, ...props }, ref) => (
  <DialogPrimitive.Overlay
    ref={ref}
    style={{
      backgroundColor: `rgba(0, 0, 0, ${backgroundOpacity})`,
    }}
    className={cn(
      `fixed inset-0 z-50   data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0`,
      className
    )}
    {...props}
  />
));
DialogOverlay.displayName = DialogPrimitive.Overlay.displayName;

interface DialogContentProps
  extends React.ComponentPropsWithoutRef<typeof DialogPrimitive.Content> {
  onClose?: () => void; // Add this line to accept an onClose prop
  showOverlay?: boolean; // Add this line to accept an showOverlay prop
  backgroundOpacity?: number;
}

const DialogContent = React.forwardRef<
  React.ElementRef<typeof DialogPrimitive.Content>,
  DialogContentProps // Use the modified props interface
>(
  (
    {
      className,
      children,
      onClose,
      showOverlay = true,
      backgroundOpacity = 0.8,
      ...props
    },
    ref
  ) => (
    <DialogPortal>
      {showOverlay && <DialogOverlay backgroundOpacity={backgroundOpacity} />}
      <DialogPrimitive.Content
        ref={ref}
        className={cn(
          "fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg",
          className
        )}
        {...props}
      >
        {children}
        <DialogPrimitive.Close
          className="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground"
          onClick={onClose} // Call the onClose function when the close button is clicked
        >
          <Cross2Icon className="h-4 w-4" />
          <span className="sr-only">Close</span>
        </DialogPrimitive.Close>
      </DialogPrimitive.Content>
    </DialogPortal>
  )
);
DialogContent.displayName = DialogPrimitive.Content.displayName;

const DialogHeader = ({
  className,
  ...props
}: React.HTMLAttributes<HTMLDivElement>) => (
  <div
    className={cn(
      "flex flex-col space-y-1.5 text-center sm:text-left",
      className
    )}
    {...props}
  />
);
DialogHeader.displayName = "DialogHeader";

const DialogFooter = ({
  className,
  ...props
}: React.HTMLAttributes<HTMLDivElement>) => (
  <div
    className={cn(
      "flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2",
      className
    )}
    {...props}
  />
);
DialogFooter.displayName = "DialogFooter";

const DialogTitle = React.forwardRef<
  React.ElementRef<typeof DialogPrimitive.Title>,
  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Title>
>(({ className, ...props }, ref) => (
  <DialogPrimitive.Title
    ref={ref}
    className={cn(
      "text-lg font-semibold leading-none tracking-tight",
      className
    )}
    {...props}
  />
));
DialogTitle.displayName = DialogPrimitive.Title.displayName;

const DialogDescription = React.forwardRef<
  React.ElementRef<typeof DialogPrimitive.Description>,
  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Description>
>(({ className, ...props }, ref) => (
  <DialogPrimitive.Description
    ref={ref}
    className={cn("text-sm text-muted-foreground", className)}
    {...props}
  />
));
DialogDescription.displayName = DialogPrimitive.Description.displayName;

export {
  Dialog,
  DialogPortal,
  DialogOverlay,
  DialogTrigger,
  DialogClose,
  DialogContent,
  DialogHeader,
  DialogFooter,
  DialogTitle,
  DialogDescription,
};

**Comment by krishna-as-dev** [comment]
in the source code `/components/ui/dialog.tsx` file make prop like `showOverlay` to DialogComponent and DialogOverlay and then you can just pass this prop from any dialog where you do not need overlay


```
const DialogOverlay = forwardRef<ElementRef<typeof DialogPrimitive.Overlay>, DialogOverlayProps>(
  ({ className, showOverlay = true, ...props }, ref) => (
    <DialogPrimitive.Overlay
      ref={ref}
      className={cn(
        'fixed inset-0 z-50  data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0',
        className,
        {
          'bg-black/80': showOverlay,
        },
      )}
      {...props}
    />
  ),
);

const DialogContent = forwardRef<
  ElementRef<typeof DialogPrimitive.Content>,
  ExtendedDialogContentProps
>(({ className, children, showOverlay, ...props }, ref) => (
  <DialogPortal>
    <DialogOverlay showOverlay={showOverlay} />
  ...rest of the component
   
```
for typescript add these types to add `showOverlay` 

```
type DialogOverlayProps = ComponentPropsWithoutRef<typeof DialogPrimitive.Overlay> & {
  showOverlay?: boolean;
};

interface ExtendedDialogContentProps
  extends ComponentPropsWithoutRef<typeof DialogPrimitive.Content> {
  showOverlay?: boolean;
}

```

now where even you want to use this just pass `showOverlay` to the DialogComponent, for eg:- 

```

  <Dialog>
     <DialogTrigger>
        <Info size={16} className='text-slate-600' />
     </DialogTrigger>
     <DialogContent showOverlay={false}>
        <DialogHeader>
          <DialogTitle>Are you absolutely sure?</DialogTitle>
              <DialogDescription>
                  This action cannot be undone. This will permanently delete your account and remove your data from our
                  servers.
              </DialogDescription>
          </DialogHeader>
    </DialogContent>
  </Dialog>
          
```
